.github/workflows/integration_test.yaml

```diff
@@ -0,0 +1,48 @@
+name: Integration tests
+
+on:
+  pull_request:
+
+jobs:
+  integration-tests:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Setup Devstack Swift
```

Comment:
Everything except this step can be run with the current pipeline. I'll try to find a way to inject steps but I don't think it'll be straightforward.
I'll keep you posted